src/main/java/formflow/library/FileController.java

```diff
@@ -50,6 +62,18 @@ public UploadController(
     this.messageSource = messageSource;
   }
 
+  /**
+   * File upload endpoint.
+   *
+   * @param file         A MultipartFile file
+   * @param flow         The current flow name, not null
```

Comment:
I think all the params are required and should not be null, so maybe add that to all of them, or none?